File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        comps = {}
        for i, n in enumerate(nums):
            if target-n in comps:
                return [comps[target-n], i]
            comps[n] = i
        return []    

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> valIdx = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) 
        {
            int compIdx;
            int comp = target - nums[i];
            if (valIdx.TryGetValue(comp, out compIdx)) {
                return new int[] { compIdx, i };
            }
            valIdx.TryAdd(nums[i], i);
        }
        return null;
    }
}